src/string.rs

```diff
@@ -0,0 +1,90 @@
+use core::slice;
+
+//Reusable type for chars, in xv6 they appear to be
+//signed chars by default
+#[allow(non_camel_case_types)]
+pub type c_char = i8;
+
+
+#[no_mangle]
+pub extern "C" fn strlen(s: *const c_char) -> isize {
+    unsafe {
+        let mut n = 0;
+        while *s.offset(n) != 0 {
+            n += 1;
+        }
+        n
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn strncmp(s: *const c_char, 
+                          t: *const c_char, 
+                          n: isize) 
```

Comment:
In the C code `n` is a `uint` should this be `n: usize`? If it is then you dont have to check `n <= 0`

Reply:
Resolved in the latest commit. Thanks for the review!